packages/jellyfish-api-core/__tests__/category/icxorderbook/listHTLCs.test.ts

```diff
@@ -0,0 +1,523 @@
+    expect(Object.keys(HTLCs).length).toBe(2) // extra entry for the warning text returned by the RPC atm.
+    expect(HTLCs[DFCHTLCTxId] as ICXDFCHTLCInfo).toStrictEqual(
+      {
+        type: ICXHTLCType.DFC,
+        status: ICXOrderStatus.OPEN,
+        offerTx: makeOfferTxId,
+        amount: DFCHTLC.amount,
+        amountInEXTAsset: DFCHTLC.amount.multipliedBy(order.orderPrice),
+        hash: DFCHTLC.hash,
+        timeout: new BigNumber(DFCHTLC.timeout as number),
+        height: expect.any(BigNumber),
+        refundHeight: expect.any(BigNumber)
+      }
+    )
+
+    // submit EXT HTLC - taker
+    const ExtHTLC: ExtHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(0.10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      htlcScriptAddress: '13sJQ9wBWh8ssihHUgAaCmNWJbBAG5Hr9N',
+      ownerPubkey: '036494e7c9467c8c7ff3bf29e841907fb0fa24241866569944ea422479ec0e6252',
+      timeout: 15
+    }
+    const ExtHTLCTxId = (await client.icxorderbook.submitExtHTLC(ExtHTLC)).txid
+    await container.generate(1)
+
+    // List htlc without limit param and check
+    const listHTLCOptionsAfterExtHTLC = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCsAfterExtHTLC: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.icxorderbook.listHTLCs(listHTLCOptionsAfterExtHTLC)
+    expect(Object.keys(HTLCsAfterExtHTLC).length).toBe(3) // extra entry for the warning text returned by the RPC atm.
+    expect(HTLCsAfterExtHTLC[ExtHTLCTxId] as ICXEXTHTLCInfo).toStrictEqual(
+      {
+        type: ICXHTLCType.EXTERNAL,
+        status: ICXOrderStatus.OPEN,
+        offerTx: makeOfferTxId,
+        amount: ExtHTLC.amount,
+        amountInDFCAsset: ExtHTLC.amount.dividedBy(order.orderPrice),
+        hash: ExtHTLC.hash,
+        htlcScriptAddress: ExtHTLC.htlcScriptAddress,
+        ownerPubkey: ExtHTLC.ownerPubkey,
+        timeout: new BigNumber(ExtHTLC.timeout),
+        height: expect.any(BigNumber)
+      }
+    )
+    expect(HTLCsAfterExtHTLC[DFCHTLCTxId] as ICXDFCHTLCInfo).toStrictEqual(
+      {
+        type: ICXHTLCType.DFC,
+        status: ICXOrderStatus.OPEN,
+        offerTx: makeOfferTxId,
+        amount: DFCHTLC.amount,
+        amountInEXTAsset: DFCHTLC.amount.multipliedBy(order.orderPrice),
+        hash: DFCHTLC.hash,
+        timeout: new BigNumber(DFCHTLC.timeout as number),
+        height: expect.any(BigNumber),
+        refundHeight: expect.any(BigNumber)
+      }
+    )
+
+    // NOTE(surangap): Enable this after AIN#599
+    // List htlc with limit of 1 and check
+    // const listHTLCOptionsWithLimit1 = {
+    //   offerTx: makeOfferTxId,
+    //   limit: 1
+    // }
+    // const HTLCsWithLimit1: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.icxorderbook.listHTLCs(listHTLCOptionsWithLimit1)
+    // expect(Object.keys(HTLCsWithLimit1).length).toBe(2)
```

Comment:
another way to write temporary skipped test in jest is to use `it.skip('', () => {})`